tests/test_cervical_cancer.py

```diff
@@ -0,0 +1,386 @@
+    """Put all people into the first stage, let progression happen (with no treatment effect) and check that people end
+    up in late stages and some die of this cause.
+    Use a healthsystem that does not allows HSI and check that diagnosis, treatment and palliative care do not occur.
+    """
+    sim = make_simulation_nohsi(seed=seed)
+
+    # set initial prevalence to be zero
+    sim = zero_out_init_prev(sim)
+
+    # no incidence of new cases
+    sim = zero_rate_of_onset_lgd(sim)
+
+    # remove effect of treatment:
+    sim = make_treatment_ineffective(sim)
+
+    # make initial population
+    sim.make_initial_population(n=popsize)
+
+    population_of_interest = get_population_of_interest(sim)
+#   sim.population.props.loc[population_of_interest, "ce_hpv_cc_status"] = 'stage1'
+    check_configuration_of_population(sim)
+
+    # Simulate
+    sim.simulate(end_date=Date(2010, 7, 1))
+    check_dtypes(sim)
+    check_configuration_of_population(sim)
+
+    df = sim.population.props
+    assert len(df.loc[df.is_alive & (df.ce_hpv_cc_status != 'none')]) > 0
+
+    # check that some people have died of cervical cancer
+    yll = sim.modules['HealthBurden'].years_life_lost
+#   todo: find out why this assert fails - I don't think it is a problem in cervical_cancer.py
+#   assert yll['CervicalCancer'].sum() > 0
+
+    # w/o healthsystem - check that people are NOT being diagnosed, going onto treatment and palliative care:
+    assert not (df.ce_date_diagnosis > start_date).any()
+    assert not (df.ce_date_treatment > start_date).any()
+    assert not (df.ce_stage_at_which_treatment_given != 'none').any()
+    assert not (df.ce_date_palliative_care > start_date).any()
+
+
+@pytest.mark.slow
+def test_check_progression_through_stages_is_blocked_by_treatment(seed):
+    """Put all people into the first stage but on treatment, let progression happen, and check that people do move into
+    a late stage or die"""
+    sim = make_simulation_healthsystemdisabled(seed=seed)
+
+    # set initial prevalence to be zero
+    sim = zero_out_init_prev(sim)
+
+    # no incidence of new cases
+    sim = zero_rate_of_onset_lgd(sim)
+
+    # remove effect of treatment:
+    sim = make_treamtment_perfectly_effective(sim)
+
+    # increase progression rates:
+    sim = incr_rates_of_progression(sim)
+
+    # make initial population
+    sim.make_initial_population(n=popsize)
+
+    # force that all persons aged over 15 are in stage 1 to begin with:
+    # get the population of interest
+    population_of_interest = get_population_of_interest(sim)
+    sim.population.props.loc[population_of_interest, "ce_hpv_cc_status"] = 'stage1'
+
+    # force that they are all symptomatic
+    sim.modules['SymptomManager'].change_symptom(
+        person_id=population_of_interest.index[population_of_interest].tolist(),
+        symptom_string='vaginal_bleeding',
+        add_or_remove='+',
+        disease_module=sim.modules['CervicalCancer']
+    )
```

Comment:
This will make all >15 yrs females be on stage 1 and have cancer symptoms yes BUT it will not automatically make everyone deemed as ever had cervical cancer in the code Hence check 
`assert set(sim.modules['SymptomManager'].who_has('vaginal_bleeding')).issubset(
      df.index[df.ce_cc_ever])` is likely to fail

Reply:
When I try to run the test file to look into this I am getting an error TypeError: 'ABCMeta' object is not subscriptable
in response to this line: from tlo import DAYS_IN_YEAR, Date, Simulation 
This was not happening before. 
Any ideas what I need to do ?

Reply:
I'm now also getting the same error when I try to run the analysis file

from tlo import Date, Simulation
TypeError: 'ABCMeta' object is not subscriptable

@thewati Not sure if you can help with this ?

Reply:
Also not sure if you @tbhallett know about this error ?

Reply:
@andrew-phillips-1 the analysis file is running on my local machine without any error. Not sure what could be wrong. I'll have to look into it

Reply:
Thanks Wati.  Let me know if it would be helpful to have a call so I can show you what I am seeing.

Reply:
@thewati Thanks Wati. Let me know if it would be helpful to have a call so I can show you what I am seeing.